fix(xen-api#_sessionOpen): remove server's credentials from login error

badrAZ opened the pull request:
xoa-support#2454

When an authentication to a server fails, the generated error will contains  the server's credentials.

This PR obfuscate these credentials.

### Check list

> Check if done, if not relevant leave unchecked.

- [x] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- [ ] documentation updated
- `CHANGELOG.unreleased.md`:
  - [x] enhancement/bug fix entry added
  - [x] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [ ] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [x] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.

badrAZ committed:
fix(xen-api#_sessionOpen): remove sensitive data from login error

xoa-support#2454

badrAZ committed:
CHANGELOG

badrAZ committed:
fix changelog

rajaa212 commented on packages/xen-api/src/index.js:
```diff
@@ -746,7 +746,10 @@ export class Xapi extends EventEmitter {
         this._interruptOnDisconnect(
-          this._call('session.login_with_password', params)
+          this._call('session.login_with_password', params).catch(error => {
+            error.call.params = '* obfuscated *'
```
What do you think about adding a comment?

julien-f commented on packages/xen-api/src/index.js:
```diff
@@ -746,7 +746,10 @@ export class Xapi extends EventEmitter {
+          this._call('session.login_with_password', params).catch(error => {
+            error.call.params = '* obfuscated *'
+            throw error
+          })
```
Wouldn't it be better to update `_call` to not add `params` if this method is called?

badrAZ replied:
By testing specifically the method name?

julien-f replied:
Yes.

badrAZ committed:
update implementation

julien-f committed:
feat(xo-server/xo.{export,import}Config}): support passphrase (#4935)

julien-f commented on CHANGELOG.unreleased.md:
```diff
@@ -17,6 +17,7 @@
 
 - Fix mounting of NFS remote in FreeBSD (PR [#4988](https://github.com/vatesfr/xen-orchestra/issues/4988))
 - [Remotes] Fix "remote is disabled" error on getting the remotes info (commit [eb2f429964d7adc264bf678c37e49a856454388e](https://github.com/vatesfr/xen-orchestra/commit/eb2f429964d7adc264bf678c37e49a856454388e))
+- Remove server's credentials from the error `SESSION_AUTHENTICATION_FAILED` (PR [#4995](https://github.com/vatesfr/xen-orchestra/pull/4995))
```
Is it really relevant to add this to the changelog?

If yes, why was it an issue that the credentials where attached to the error?

badrAZ replied:
Password attempts can be personal passwords, other servers' password it should not be logged/viewed by other system admins.

julien-f replied:
Your changelog entry does not mention logging.

badrAZ committed:
changelog

badrAZ committed:
Merge branch 'master' into obfuscateServersCredentials

julien-f commented on CHANGELOG.unreleased.md:
```diff
@@ -19,6 +19,7 @@
 - Fix mounting of NFS remote in FreeBSD (PR [#4988](https://github.com/vatesfr/xen-orchestra/issues/4988))
 - [Remotes] Fix "remote is disabled" error on getting the remotes info (commit [eb2f429964d7adc264bf678c37e49a856454388e](https://github.com/vatesfr/xen-orchestra/commit/eb2f429964d7adc264bf678c37e49a856454388e))
 - Fix default filters not being set in all tables (PR [#4994](https://github.com/vatesfr/xen-orchestra/pull/4994))
+- Don't log server's credentials in case of the error `SESSION_AUTHENTICATION_FAILED` (PR [#4995](https://github.com/vatesfr/xen-orchestra/pull/4995))
```
```suggestion
- Don't log server's credentials in case of `SESSION_AUTHENTICATION_FAILED` error (PR [#4995](https://github.com/vatesfr/xen-orchestra/pull/4995))
```

benjamreis committed:
fix(sdn-controller): recreate both sides of tunnels when host updated (#4996)

To keep password synchronized.

pdonias committed:
feat(xo-web/modal/form): don't close on click or escape (#5002)

See xoa-support#2469

When the user clicks outside of a form modal or presses escape, don't close the
modal as all the information they entered so far will be lost.

julien-f committed:
fix(xo-server-auth-ldap): dont pass `bindDN` and `bindCredentials to LDAP client

It was left from an older version of the code which was working differently.

julien-f committed:
feat(xo-server-auth-ldap): support startTLS

Fixes #4999

julien-f committed:
fix(xo-server-auth-ldap): upgrade ldapts dep

Necessary for 5870f6f73

Also, only pass `tlsOptions` when creating client when using `ldaps`.

julien-f committed:
fix(xo-server-auth-ldap): use `this._tlsOptions` in `bind`

Introduced by 5870f6f73

badrAZ committed:
Update CHANGELOG.unreleased.md

Co-authored-by: Julien Fontanet <julien.fontanet@isonoe.net>

badrAZ committed:
Merge branch 'master' into obfuscateServersCredentials
